modules/generator/processor/servicegraphs/servicegraphs.go

```diff
@@ -188,29 +196,51 @@ func (p *processor) consume(resourceSpans []*v1.ResourceSpans) error {
 	return nil
 }
 
-func (p *processor) Shutdown(ctx context.Context) {
+func (p *processor) Shutdown(_ context.Context) {
 	close(p.closeCh)
 }
 
 // collectEdge records the metrics for the given edge.
 // Returns true if the edge is completed or expired and should be deleted.
 func (p *processor) collectEdge(e *store.Edge) {
 	if e.IsCompleted() {
-		clientServerLabelValues := registry.NewLabelValues([]string{e.ClientService, e.ServerService})
+		values := []string{e.ClientService, e.ServerService}
+		for _, dimension := range p.cfg.Dimensions {
+			values = append(values, e.Dimensions[dimension])
+		}
+		labelValues := registry.NewLabelValues(values)
 
-		p.serviceGraphRequestTotal.Inc(clientServerLabelValues, 1)
+		p.serviceGraphRequestTotal.Inc(labelValues, 1)
 		if e.Failed {
-			p.serviceGraphRequestFailedTotal.Inc(clientServerLabelValues, 1)
+			p.serviceGraphRequestFailedTotal.Inc(labelValues, 1)
 		}
 
-		p.serviceGraphRequestServerSecondsHistogram.Observe(clientServerLabelValues, e.ServerLatencySec)
-		p.serviceGraphRequestClientSecondsHistogram.Observe(clientServerLabelValues, e.ClientLatencySec)
+		p.serviceGraphRequestServerSecondsHistogram.Observe(labelValues, e.ServerLatencySec)
+		p.serviceGraphRequestClientSecondsHistogram.Observe(labelValues, e.ClientLatencySec)
 	} else if e.IsExpired() {
-		p.metricUnpairedEdges.Inc()
+		p.metricExpiredSpans.Inc()
+	}
+}
+
+func (p *processor) upsertDimensions(m map[string]string, resourceAttr []*v1common.KeyValue, attr []*v1common.KeyValue) {
+	for _, dim := range p.cfg.Dimensions {
+		for _, kv := range resourceAttr {
+			key := kv.Key
+			if key == dim {
+				m[key] = commonUtil.StringifyAnyValue(kv.Value)
+				continue
```

Comment:
This `continue` should break to the outer loop since we found the dimension, right?
```go
for _, dim := range p.cfg.Dimensions {
```
Maybe add a label to clarify what for-loop we want to continue.

Reply:
Refactored in 309793e3fe14e9817d916b0c89d46c77f40508d5